controllers/swiftproxy_controller.go

```diff
@@ -429,6 +430,15 @@ func (r *SwiftProxyReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	envVars[sps.Name] = env.SetValue(hash)
 	password := string(sps.Data[instance.Spec.PasswordSelectors.Service])
 
+	secretRef := ""
+	if instance.Spec.EncryptionEnabled {
+		secretRef, err = swiftproxy.GetBarbicanSecret(instance, helper, keystonePublicURL, password)
+		if err != nil {
+			r.Log.Error(err, "Failed to get secretRef from Barbican")
+			return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil
+		}
+	}
```

Comment:
It might be good to have some condition logic here, similar to L414-428

Reply:
thx, done!